### ISSUE TITLE
Include hunspell dictionaries in org.gnome.Platform.Locale

### ISSUE THREAD

**Issue by stbergmann** [flatpak/flatpak#702]
In the GNOME runtime (as advertised at <http://flatpak.org/runtimes.html>), hunspell_en data is available as /usr/share/myspell/en_*.* in the runtime (presumably coming directly from org.gnome.Platform, not org.gnome.Platform.Locale), but no other locales.  Would be good (e.g., for spell-checking support in the LibreOffice flatpak) if data for all relevant locales was available in org.gnome.Platform or org.gnome.Platform.Locale.  (Tested with 3.20 and 3.22.)

**Comment by nesnera** [comment]
Good news. Thanks!
(BTW - my preferred DE is [LXQt](http://lxqt.org/). I know - it's another cup of tea but evangelizacion must be. O:-) )

**Comment by nesnera** [comment]
Manual workaround:
1) get [Hunspell](https://en.wikipedia.org/wiki/Hunspell) dictionaries for you language (e.g. at [Fedora](https://en.wikipedia.org/wiki/Fedora) for Czech)
   [dnf](https://en.wikipedia.org/wiki/DNF_(software)) install hunspell-cs
2) dictionary files are placed into the /usr/share/myspell/cs_  Copy (or symlink) them into appropriate myspell flatpack folder. Find it as
find ~ -iname myspell
and
cp /usr/share/myspell/cs_* flatpackPath/files/share/myspell/
3) just run LibreOffice and set menu Tools > Options... > Language Settings > Languages

**Comment by gnumdk** [comment]
Argh :-1: This is really needed, flatpak is really useless if this is not working... Bad news, need to package my app to Fedora/OpenSUSE/Debian/Ubuntu :-(

**Comment by alexlarsson** [comment]
I agree that we should put the dictionaries in the locales. Does anyone want to work on this?

**Comment by gnumdk** [comment]
@alexlarsson  me as I really need it :)

**Comment by gnumdk** [comment]
Can't find org.gnome.Platform.json and org.gnome.Platform.Locale.json

**Comment by alexlarsson** [comment]
First of all, the dictionaries should be in the freedesktop.org runtime, so they work for everyone, not only gnome. That is available here: https://github.com/flatpak/freedesktop-sdk-images

Secondly, there is no separate json for the platform, or the locale. What happens is that we build the Sdk, and then we create the platform from that by reoving things we don't need. The locale is automatically extracted from what is in /usr/share/runtime/locale when the sdk is built, so all you have to do is install the locale data in the right subdirectory of that (say /usr/share/runtime/locale/cs/ for czech) and it will be auto-installed by people using the cs locale.

**Comment by gnumdk** [comment]
After looking at Makefile-hunspell-en, seems dict are installed in /usr/share/myspell. 
Putting them in /usr/share/runtime/locale/fr (for example) will not work I guess. Do I need to modify something elsewhere?

**Comment by gnumdk** [comment]
Hello, when looking at #924, I see that for fonts, flatpak is just mapping system directory.

Why not just do that for dictionaries? Instead of having to installed version (system one and flatpak one).

**Comment by gnumdk** [comment]
BTW, I'm not able to make GtkSpell look for another directory than /usr/share/myspell...

**Comment by TingPing** [comment]
@gnumdk So there are multiple layers of indrection going on there, gtkspell uses enchant which afaik exposes no way to modify dictionary paths even if hunspell directly does (which I haven't looked up). Can you confirm that a custom hunspell build in `/app` works?

**Comment by mcatanzaro** [comment]
Wow apparently enchant is maintained again. Based on the README, it also reads dictionaries out of ~/.config: https://github.com/AbiWord/enchant

**Comment by gnumdk** [comment]
@TingPing I added a hunspell.tgz (all dicts from Fedora 28) to my flatpak app (oversized).
```

{
      "name": "myspell",
      "buildsystem": "simple",
       "build-commands": [
            "mkdir /app/share/myspell",
            "cp * /app/share/myspell"
      ],


```

**Comment by aldevel** [comment]
Is hunspell implemented somewhere or not yet? Asking because it still won't work with my installation.

**Comment by TingPing** [comment]
@aldevel 	Sounds like a Libreoffice bug, the dictionaries are in the runtime and work fine in other applications here.

**Comment by aldevel** [comment]
I think one of us just misunderstood something. I mean am I right this thread about hunspell in Flatpak? Do you mentioning what kind of runtime and where? I have to note the system wide hunspell installation what works for me well is absolutely not related to this issue, at least as I see there is no way to make it work with Flatpaks Libreoffice for example. Please help me out here what's going on or just wording more clearly please, thanks.

**Comment by stbergmann** [comment]
@aldevel, LibreOffice isn't even using the org.gnome.Platform mentioned in this issue any more, so if you have an issue with LibreOffice please file a fresh issue at <https://github.com/flathub/org.libreoffice.LibreOffice/issues>, stating exactly which version of LibreOffice you are using (the "Version: ..." and following lines in the "About LibreOffice" dialog accessed through menu entry "Help - About LibreOffice" can be copied; please paste all of them into the issue), what exactly you are doing, and what exactly does not work as expected.

**Comment by llabusch93** [comment]
Any news on this here?

I want to use Evolution as a Flatpak, but I need the German spell checking in there...

**Comment by dchristidis** [comment]
@llabusch93 In case this proves helpful to you, I’ve configured `extra-languages` as described [here](https://docs.flatpak.org/en/latest/flatpak-command-reference.html?highlight=extra-languages#flatpak-config) and run `flatpak update`. Then, the desired dictionaries were made available in Evolution.

**Comment by Mikenux** [comment]
@TingPing: Is there anything else left to do?

**Comment by bbhtt** [comment]
if there is still a bug here, it should be reported to the runtimes. https://gitlab.com/freedesktop-sdk/freedesktop-sdk/-/issues https://gitlab.gnome.org/GNOME/gnome-build-meta/-/issues
